### DISCUSSION TITLE
Can a getter be used in a query condition?

### DISCUSSION THREAD

**Discussion by vineetdigit** [Automattic/mongoose#10914] [Q&A] [answered]
In the following example, 'type' is a getter defined on the Post Schema. 'type' is of type string that the getter computes after looking at one or more fields of the document. Even though the database contains a post document with valid postId that will compute type to "comment", the following query always returns null. I'm wondering if mongoose supports using a getter in the query document?

```
let post = await Post.findOneAndUpdate(
      { postId: postId, type: "comment" },
      { data: newData },
      { new: true }
    );
```

**Comment by vkarpov15** [comment, marked as answer]
Unfortunately not. Since virtuals aren't stored in MongoDB, you can't use virtuals in query filters.

**Reply by vineetdigit** [reply]
Thank you!
